src/services/PostEventProcessor.ts

```diff
@@ -0,0 +1,133 @@
+import { Logger } from "@aws-lambda-powertools/logger";
+import { Metrics } from "@aws-lambda-powertools/metrics";
+
+import { AppError } from "../utils/AppError";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { Constants } from "../utils/Constants";
+import { IPRService } from "./IPRService";
+import { EnvironmentVariables } from "./EnvironmentVariables";
+import { ServicesEnum } from "../models/enums/ServicesEnum";
+import { createDynamoDbClient } from "../utils/DynamoDBFactory";
+import { IpvStartedOnEvent, JourneyWentAsyncOnEvent, ReadyToResumeOnEvent, AccountDeletedOnEvent } from "../models/IPREventTypes";
+
+
+export class PostEventProcessor {
+	private static instance: PostEventProcessor;
+
+	private readonly logger: Logger;
+
+	private readonly metrics: Metrics;
+
+	private readonly environmentVariables: EnvironmentVariables;
+
+	private readonly iprService: IPRService;
+
+	constructor(logger: Logger, metrics: Metrics) {
+		this.logger = logger;
+		this.metrics = metrics;
+		this.environmentVariables = new EnvironmentVariables(logger, ServicesEnum.POST_EVENT_SERVICE);
+		this.iprService = IPRService.getInstance(this.environmentVariables.sessionEventsTable(), this.logger, createDynamoDbClient());
+	}
+
+	static getInstance(logger: Logger, metrics: Metrics): PostEventProcessor {
+		if (!PostEventProcessor.instance) {
+			PostEventProcessor.instance = new PostEventProcessor(logger, metrics);
+		}
+		return PostEventProcessor.instance;
+	}
+
+	async processRequest(eventBody: any): Promise<any> {
+		try {
+			const eventDetails = JSON.parse(eventBody);
```

Comment:
Can we validate the event matches one of our expected interfaces? Or is thing being done elsewhere?

Reply:
It's done else where but I think we should validate here too so we catch invalid payloads early, will update